packages/medusa/src/api/routes/admin/order-edits/create-order-edit.ts

```diff
@@ -0,0 +1,81 @@
+import { Request, Response } from "express"
+import { OrderEditService } from "../../../../services"
+import { IsOptional, IsString } from "class-validator"
+import { EntityManager } from "typeorm"
+
+/**
+ * @oas [post] /order-edits
+ * operationId: "PostOrderEdits"
+ * summary: "Create an OrderEdit"
+ * description: "Created a OrderEdit."
+ * x-authenticated: true
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.admin.orderEdit.create({ order_id, internal_note })
+ *       .then(({ order_edit }) => {
+ *         console.log(order_edit.id);
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request POST 'https://medusa-url.com/admin/order-edits' \
+ *       --header 'Authorization: Bearer {api_token}'
+ *       -d '{ "order_id": "my_order_id", "internal_note": "my_optional_note" }'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - OrderEdit
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *             order_edit:
+ *               $ref: "#/components/schemas/order_edit"
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ *   "401":
+ *     $ref: "#/components/responses/unauthorized"
+ *   "404":
+ *     $ref: "#/components/responses/not_found_error"
+ *   "409":
+ *     $ref: "#/components/responses/invalid_state_error"
+ *   "422":
+ *     $ref: "#/components/responses/invalid_request_error"
+ *   "500":
+ *     $ref: "#/components/responses/500_error"
+ */
+export default async (req: Request, res: Response) => {
+  const orderEditService = req.scope.resolve(
+    "orderEditService"
+  ) as OrderEditService
+  const manager = req.scope.resolve("manager") as EntityManager
+
+  const data = req.validatedBody as AdminPostOrderEditsReq
+  const loggedInUserId = (req.user?.id ?? req.user?.userId) as string
+
+  const orderEdit = await manager.transaction(async (transactionManager) => {
+    return await orderEditService
+      .withTransaction(transactionManager)
+      .create(data, { loggedInUserId })
+  })
+
```

Comment:
But at that point we did not add any changes, it is a raw order edit so i don’t think that computed the line items would had something no?

Reply:
at the moment the compute line items would not return any lines, maybe I should update the implementation so that it takes into account that point?

Reply:
is it something like that that we expect to have https://github.com/medusajs/medusa/pull/2187/commits/9cb7b3ed6fe00d406ae1643d3b9269ad29fddf6c ? Basically if there is an update, we attach the data of the updated line item otherwise the data of the original one if none exists

Reply:
for the totals, we would need to re compute the tax_lines as well?

Reply:
It should return the items on the order right?

The idea here is that we give the consumer they need to make changes without having to fetch additional resources

Reply:
which is what we agreed on [here ](https://www.notion.so/medusajs/Order-Editing-6939ab89af9948688968c9387f98bab3#362e5e15c71648a785c14b1a8760be67) with some changes to the pseudo code to work. it is an aggregate that include the changes made to the line items so that with the id if the user want to update it again we can retrieve the changes with the original_line_item_id for an update and line_item_id for a create

Reply:
I ve added a comment on the method to describe what it does :)

Reply:
Alright might have been tooo fast then - the `items` fields should contain the items as they will be when the order edit is confirmed

Reply:
Yes this is the case minus the id which correspond to the id of the original line, but at for now we don’t have the line generated so we don’t have the tax lines and adjustments etc

Reply:
So basically all the items that have not been touched from the order will be there plus those updated and added

Reply:
For now after the creation, the totals will be the same as the order since no item have been altered yet